src/yb/integration-tests/cluster_itest_util.cc

```diff
@@ -878,7 +882,8 @@ Status GetReplicaStatusAndCheckIfLeader(const TServerDetails* replica,
   const string& replica_uuid = replica->instance_id.permanent_uuid();
   if (cstate.has_leader_uuid() && cstate.leader_uuid() == replica_uuid &&
       (lease_check_mode == LeaderLeaseCheckMode::DONT_NEED_LEASE ||
-       leader_lease_status == consensus::LeaderLeaseStatus::HAS_LEASE)) {
+       leader_lease_status == consensus::LeaderLeaseStatus::HAS_LEASE) && 
+       (!(check_no_op_committed) || (check_no_op_committed && leader_no_op_committed_))) {
```

Comment:
This condition is equivalent to `!check_no_op_committed || leader_no_op_committed_`